doc/build.jl

```diff
@@ -0,0 +1,403 @@
+using LightGraphs
+
+# This file generated the Markdown documentation files.
+
+# The @file macro generates the documentation for a particular file
+# where the {{method1, methods2}} includes the documentation for each method
+# via the `buildwriter` function.
+
+# Currently this prints the methodtable followed by the docstring.
+
+macro file(args...) buildfile(args...) end
+
+buildfile(t, s::AbstractString) = buildfile(t, Expr(:string, s))
+
+buildfile(target, source::Expr) = quote
+    open(joinpath(dirname(@__FILE__), $(esc(target))), "w") do file
+        println(" - '$($(esc(target)))'")
+        println(file, "<!-- AUTOGENERATED. See 'docs/build.jl' for source. -->")
+        $(Expr(:block, [buildwriter(arg) for arg in source.args]...))
+    end
+end
+
+buildwriter(ex::Expr) = :(print(file, $(esc(ex))))
+
+buildwriter(t::AbstractString) = Expr(:block,
+    [buildwriter(p, iseven(n)) for (n, p) in enumerate(split(t, r"{{|}}"))]...
+)
+
+buildwriter(part, isdef) = isdef ?
+    begin
+        parts = Expr(:vect, [:(($(parse(p))), @doc($(parse(p)))) for p in split(part, r"\s*,\s*")]...)
+        quote
+            for (f, docstring) in $(esc(parts))
+                if isa(f, Function)
+                    md_methodtable(file, f)
+                end
+                writemime(file, "text/plain", docstring)
+                println(file)
+            end
+        end
+    end :
+    :(print(file, $(esc(part))))
+
+function md_methodtable(io, f)
+    println(io, "### ", first(methods(f)).func.code.name)
+    println(io, "```")
+    for m in methods(f)
+        md_method(io, m)
+    end
+    println(io, "```")
+end
+function md_method(io, m)
+    # We only print methods with are defined in the parent (project) directory
+    if !(startswith(string(m.func.code.file),
+                    splitdir(dirname(@__FILE__))[1]))
```

Comment:
This bit. If your LightTables is being imported from a different place that's gonna fail.

We could test the file of LightTables, in fact that's probably better than this!